custom/icds_reports/utils/aggregation_helpers/distributed/agg_child_health.py

```diff
@@ -149,47 +194,78 @@ def aggregation_query(self):
             else:
                 name = c[0]
             query_cols.append((name, c[1]))
-        return """
-        CREATE UNLOGGED TABLE "{tmp_tablename}" AS SELECT
-            {query_cols}
-            FROM "{child_health_monthly_table}" chm
-            LEFT OUTER JOIN "awc_location" awc_loc ON awc_loc.doc_id = chm.awc_id
-            WHERE chm.month = %(start_date)s AND awc_loc.state_id != '' AND awc_loc.state_id IS NOT NULL
-            GROUP BY awc_loc.state_id, awc_loc.district_id, awc_loc.block_id, awc_loc.supervisor_id, chm.awc_id,
-                     chm.month, chm.sex, chm.age_tranche, chm.caste,
-                     coalesce_disabled, coalesce_minority, coalesce_resident
-            ORDER BY awc_loc.state_id, awc_loc.district_id, awc_loc.block_id, awc_loc.supervisor_id, chm.awc_id;
-        INSERT INTO "{tablename}" ({final_columns}) SELECT * from "{tmp_tablename}";
-        DROP TABLE "{tmp_tablename}";
-        """.format(
-            tablename=self.tablename,
-            final_columns=", ".join([col[0] for col in columns]),
-            query_cols=", ".join(['{} as {}'.format(q, name) for name, q in query_cols]),
-            child_health_monthly_table='child_health_monthly',
-            tmp_tablename='tmp_{}'.format(self.tablename)
-        ), {
-            "start_date": self.month
-        }
+
+        query_cols = ", ".join([f'{q} as {name}' for name, q in query_cols])
+        child_health_monthly_table = 'child_health_monthly'
+        tmp_tablename = 'blah_blah_blah'
+        final_columns = ", ".join([col[0] for col in columns])
+        supervisor_id_ranges = [
+            ('0', '1'),
+            ('1', '2'),
+            ('2', '3'),
+            ('3', '4'),
+            ('4', '5'),
+            ('5', '6'),
+            ('6', '7'),
+            ('7', '8'),
+            ('8', '9'),
+            ('9', 'A'),
+            ('A', 'B'),
+            ('B', 'C'),
+            ('C', 'D'),
+            ('D', 'E'),
+            ('E', 'F'),
+            ('F', 'a'),
+            ('a', 'b'),
+            ('b', 'c'),
+            ('c', 'd'),
+            ('d', 'e'),
+            ('e', 'f'),
+            ('f', 'zzzzzzzzzz'),
```

Comment:
Ideally there would be a `< '0'` query and an `> 'f'` query, but this covers all real cases, its been a long day and its getting late and I wanted to watch bake off before I go to bed.

Reply:
:cake: